apps/system/utils/netcmd.c

```diff
@@ -95,6 +95,95 @@ struct ifconfig_cmd_info_s {
 extern int netdb_main(int argc, char *argv[]);
 #endif
 
+int getIp(void)
+{
+	struct ifreq *ifr;
+	struct sockaddr_in *sin;
+	struct sockaddr *sa;
+	struct ifconf ifcfg;
+	int ret = -1;
+	int fd;
+	int numreqs = 3;
+	int num_nic = 0;
+	int ret1 = 0;
+	struct ifaddrs *ifa = NULL, *ifp = NULL;
+	ret = netlib_getifaddrs(&ifa);
+	if (ret < 0) {
+		return 0;
+	}
+
+	fd = socket(AF_INET, SOCK_DGRAM, 0);
+	if (fd < 0) {
+		return 0;
+	}
+	memset(&ifcfg, 0, sizeof(ifcfg));
+	ifcfg.ifc_buf = NULL;
+	ifcfg.ifc_len = sizeof(struct ifreq) * numreqs;
+	ifcfg.ifc_buf = malloc(ifcfg.ifc_len);
+	if (ioctl(fd, SIOCGIFCONF, (unsigned long)&ifcfg) < 0) {
+		perror("SIOCGIFCONF ");
+		goto DONE;
+	}
+	num_nic = ifcfg.ifc_len / sizeof(struct ifreq);
+	ifr = ifcfg.ifc_req;
+	int i = 0;
+
+	for (; i < num_nic; ifr++, i++) {
+		sin = (struct sockaddr_in *)&ifr->ifr_addr;
+		if ((sin->sin_addr.s_addr) == INADDR_LOOPBACK) {
+		} else {
+			struct ifreq tmp;
+			strncpy(tmp.ifr_name, ifr->ifr_name, IF_NAMESIZE);
+			ret = ioctl(fd, SIOCGIFHWADDR, (unsigned long)&tmp);
+			if (ret < 0) {
+			} else {
+				sa = &tmp.ifr_hwaddr;
+			}
+
+			ret = ioctl(fd, SIOCGIFFLAGS, (unsigned long)ifr);
+			if (ret < 0) {
+			} else {
+			}
+		}
+					
+		ret = ioctl(fd, SIOCGIFNETMASK, (unsigned long)ifr);
+		if (ret < 0) {
+			
+		} else {
+			sin = (struct sockaddr_in *)&ifr->ifr_addr;
+			
+		}
+
+		ret = ioctl(fd, SIOCGIFMTU, (unsigned long)ifr);
+		if (ret < 0) {
+			
+		} else {
+
+		}
+
+		for (ifp = ifa; ifp; ifp = ifp->ifa_next) {
+			if (ifp->ifa_addr && ifp->ifa_addr->sa_family == AF_INET6
+				&& !strncmp(ifr->ifr_name, ifp->ifa_name, IFNAMSIZ)) {
+				struct sockaddr_in6 *sin6 = (struct sockaddr_in6 *)ifp->ifa_addr;
+				char ipaddr[INET6_ADDRSTRLEN + 1] = {0,};
+				memset(ipaddr, 0, INET6_ADDRSTRLEN + 1);
+				inet_ntop(AF_INET6, (void *)&(sin6->sin6_addr), ipaddr, INET6_ADDRSTRLEN);
+				
+				if(strcmp(ipaddr,"::1")!=0){
```

Comment:
For security issue, it is better to use `strncmp` instead of `strcmp`.
Please check this :)